api/v1alpha1/addon_types.go

```diff
@@ -115,6 +115,12 @@ type CNI struct {
 	// +kubebuilder:default=HelmAddon
 	// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
 	Strategy *AddonStrategy `json:"strategy,omitempty"`
+
+	// ValuesRef is an object reference to Configmap/Secret inside the same namespace
+	// which contains inline YAML representing the values for the Helm chart
+	// when HelmAddon is the strategy.
```

Comment:
I think CRS strategy is going to be deprecated eventually. Should we assume that `valuesRef` always used for helm strategy and mentioning `the values for the Cilium Helm chart` in the field description is good enough.
```suggestion
	// ValuesRef is an object reference to Configmap/Secret inside the same namespace
	// which contains inline YAML representing the values for the Cilium Helm chart
```

Reply:
don't think it's much of concern right now.. Once we deprecate CRS from [AddonStrategy](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1011/files#diff-6fd54619366c9d6ade856bbf25da675fea8f7d9384b0a947ea742b637d66cc7bR116), we could always update the `ValuesRef` description here. Are your concerned that we'll forget this later?

Reply:
Thanks for making the change. my only concern was that description refers to other field in the schema but we do not strictly validate it in the code. So we can relax the description a bit.
OR
We can also validate in the webhook if ValuesRef is used with wrong strategy.

Reply:
Intention is to inform customer when `ValuesRef` field will be considered, API description is a low hanging fruit. Later, webhook checks can also be introduced if we want to make API input restrictive and otherwise it would be a no-op even if `ValuesRef` is specified as per description. If we don't add it in API description or webhook then, users wouldn't know the field usage.